lnwallet/mock.go

```diff
@@ -0,0 +1,351 @@
+package lnwallet
+
+import (
+	"encoding/hex"
+	"sync/atomic"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/btcutil/hdkeychain"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	base "github.com/btcsuite/btcwallet/wallet"
+	"github.com/btcsuite/btcwallet/wallet/txauthor"
+	"github.com/btcsuite/btcwallet/wtxmgr"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+var (
+	CoinPkScript, _ = hex.DecodeString("001431df1bde03c074d0cf21ea2529427e1499b8f1de")
+)
+
+// mockWalletController is a mock implementation of the WalletController
+// interface. It let's us mock the interaction with the bitcoin network.
+type mockWalletController struct {
```

Comment:
Non-blocking, just that I think it's better for us to use `mock.Mock` as the current approach makes us reimplement the interface, and this can get complicated [quickly](https://github.com/lightningnetwork/lnd/blob/master/routing/mock_test.go#L27-L77). I'm trying something new [here](https://github.com/lightningnetwork/lnd/pull/6683/commits/47207dc76217b8bb9162318d7b5261bcad93cadd#diff-4963b0f520937f7e5a4aa825c3541d17af3514a2c7d8f94ce83ccbe4308b0739R741-R776). And once we are comfortable with it, I'll create followup PRs to change these plus moving commonly used mocks inside `lnmock`.

Reply:
Yeah copy/paste def isn't the best solution. I think the gap in my mental model re mocks and Go is: without strong duck typing or monkey patching, must we generated compiled instances of all the mocks we care about?